nextjs-13/src/app/api/v1/newsletter/[id]/route.test.ts

```diff
@@ -49,3 +53,53 @@ describe('DELETE /v1/newsletter/:id', () => {
     });
   });
 });
+
+describe('PUT /v1/newsletter/:id', () => {
+  const user = { ...userFactory, id: '1' };
+  const newsletter = { id: '2' };
+  const data = {
+    name: 'New name',
+    content: 'New content',
+  };
+  const where = {
+    id: newsletter.id,
+    user: user,
+  };
+  const requestBody = { json: () => data };
+  const args = { where, data };
+
+  beforeEach(() => {
+    appHandler.mockImplementation((_, callback) => callback(user, {}));
+    updateNewsletter.mockResolvedValue({ count: 1 });
+  });
```

Comment:
For the test above in `DELETE`, the usage of `beforeEach` was more minimal, but this implementation for `PU` is abusing too much on shared fixtures which is an anti-pattern: https://github.com/nimblehq/compass/pull/1274/files#diff-bc1c8aea77bf01da18ac0ba6c224d40694f9c88fa9a6f7d2a574567372ae884cR105. Fixtures should not be shared.

It is better to repeat the fixture creation in each test as 1) it makes the test easier to read, 2) it ensures we only create the required fixtures for each test, and 3) debugging is easier.

So the following is expected:

```javascript
describe('newsletter exists', () => {
    it('updates the newsletter', async () => {
      const user = { ...userFactory, id: '1' };
      const newsletter = { id: '2' };
      const data = {
        name: 'New name',
        content: 'New content',
      };
      const where = {
        id: newsletter.id,
        user: user,
      };

     appHandler.mockImplementation((_, callback) => callback(user, {}));
     updateNewsletter.mockResolvedValue({ count: 1 });

      const response = await PUT({ json: () => data }, {
         params: { id: newsletter.id },
      });

      expect(updateNewsletter).toHaveBeenCalledWith({ where, data });
      expect(response.status).toBe(StatusCodes.OK);
    });
});

describe('newsletter does NOT exist', () => {
   it('returns error', async () => {
      const user = { ...userFactory, id: '1' };
      const newsletter = { id: '2' };

      appHandler.mockImplementation((_, callback) => callback(user, {}));
      updateNewsletter.mockResolvedValue({ count: 0 });

      const response = await PUT({ json: () => {}, {
        params: { id: newsletter.id },
      });
      const responseBody = await response.json();

      expect(response.status).toBe(StatusCodes.UNPROCESSABLE_ENTITY);
      expect(responseBody).toMatchObject({
        message: 'Newsletter could not be updated',
      });
   });
});
```

Reply:
Ohh I see.

So these fixtures are shared here, instead of with Rspec, the default behavior of factories(using like `let(:user) { User.create }`) are not shared. I thought they also don't share here.

Oke let me improve this.

Reply:
Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/117/commits/bfb0be0590523740e6bf06ff7594d02cc26624a1#diff-e9a6bb57657e77990e30f03a0bd834fe36456159e16ea91f33c4bc2de833cf31R58-R109)